python/GafferUI/VectorDataWidget.py

```diff
@@ -1338,6 +1381,113 @@ def _createEditorInternal( self, index ) :
 _Delegate.registerType( IECore.QuatfVectorData.staticTypeId(), _NumericDelegate )
 _Delegate.registerType( IECore.QuatdVectorData.staticTypeId(), _NumericDelegate )
 
+# A delegate that creates a `_NumericDelegate` for each of the color components
+# and an editable color swatch for the color column.
+class _ColorDelegate( _Delegate ) :
+
+	def __init__( self ) :
+
+		_Delegate.__init__( self )
+
+		self.__colorChooser = None
+		self.__popup = None
+
+	def paint( self, painter, option, index ) :
+
+		isColor, value = self.__colorData( index )
+
+		if isColor :
+
+			# in PyQt, option is passed to us correctly as a QStyleOptionViewItemV4,
+			# but in PySide it is merely a QStyleOptionViewItem and we must "cast" it.
+			if hasattr( QtGui, "QStyleOptionViewItemV4" ) :
+				option = QtGui.QStyleOptionViewItemV4( option )
+
+			transformedColor = GafferUI.DisplayTransform.get()( value )
+
+			backgroundBrush = QtGui.QBrush(
+				QtGui.QColor(
+					transformedColor.r * 255.0,
+					transformedColor.g * 255.0,
+					transformedColor.b * 255.0,
+					255.0 if isinstance( value, imath.Color3f ) else value.a * 255.0
+				)
+			)
+
+			painter.fillRect( option.rect, backgroundBrush )
+		else :
+			_Delegate.paint( self, painter, option, index )
+
+	def createEditor( self, parent, option, index ) :
+
+		w = _Delegate.createEditor( self, parent, option, index )
+
+		isColor, value = self.__colorData( index )
+		if isColor :
+			w.setParent( None )
+			# Restore popoup flag after `setParent()` above
+			w.setWindowFlags( QtCore.Qt.Popup )
+
+		return w
```

Comment:
It seems a bit awkward that we have to override this - perhaps it makes sense for `_Delegate.createEditor()` to skip the `setParent()` call when it sees that the editor is a window? That would also provide more flexibility to people using `editSignal()` to make an editor.